### ISSUE TITLE
Bug with namespace function + AMD

### ISSUE THREAD

**Issue by bmaland** [gruntjs/grunt-contrib-handlebars#108]
I'm using a namespace function which appends the folder name to the template namespace, resulting in the following namespaces: JST.shared/JST.module1/JST.module2. This works fine without AMD.

The problem is that when AMD is enabled, the last namespace "wins" because of the following code: `output.push("return "+nsInfo.namespace+";");`  `nsInfo` is here modified in the forEach loop, which leaves it at JST.module2 in my case. Alas, only JST.module2 is returned from the compiled template file instead of the root JST namespace.

I'm not sure how to fix this other than adding a new configuration option. Guessing the root namespace seems hard (you can't simply remove all dots), but maybe I'm missing something.

**Comment by stephanebachelier** [comment]
@bmaland it's merged in v0.9.2

**Comment by lostthetrail** [comment]
Just a side note, this is a backwards incompatible bug fix.

The precompile templates file now generated by a few of my projects were coded to return the specific namespace configured in options, but now it returns the lowest level namespace.

Before 0.9.2
```
return this["JS"]["templates"]["foo"];
```

After 0.9.2
```
return this["JS"];
```

All of the templates in this file belong to JS.templates.foo.

The issue being reported is valid, but I don't know that this solution is complete.

"The namespace in which the precompiled templates will be assigned."

If I assign namespace to "JS.templates.foo", the nsDeclarations will show 3 declarations, but the new 'extractGlobalNamespace' function will only get "JS", which isn't correct for the configured namespace.

**Comment by stephanebachelier** [comment]
@lostthetrail test my branch in #131 as it should fixes your bug.

**Comment by lostthetrail** [comment]
@stephanebachelier This looks like the correct solution. I will check out your branch and test it tomorrow morning. Thank you for responding quickly to the issue.

**Comment by lostthetrail** [comment]
Ah drat. I noticed something that might still be problematic.

If you have a function that returns a dot based namespace, you will still have the issue.

```
var myVar = 'FooVar';

namespace: function() {
    return 'Foo.Templates.' + myVar;
}
```

**Comment by stephanebachelier** [comment]
@lostthetrail I will try to update my fix #131 with this case
